From 422bfb86ddc9ef265f0de2bff9339e8b05fc9f90 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Fri, 15 Oct 2021 16:16:15 +0800
Subject: [PATCH] Revert "[Huawei] block: avoid creating invalid symlink file
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 422bfb86ddc9ef265f0de2bff9339e8b05fc9f90

 for patitions"

hulk inclusion
category: bugfix
bugzilla: 55097 https://gitee.com/openeuler/kernel/issues/I4DDEL

-------------------------------------------------

This reverts commit 6afb4716beff7102784a06fda7df7cd703721a8d.

The patch set for partition symlink cleanup will introduce
deadlock for nbd, loop and xen-blkfront driver, so revert it.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
Reviewed-by: Hou Tao <houtao1@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 block/genhd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/genhd.c b/block/genhd.c
index 5a7196286241..b6e7c00c384a 100644
--- a/block/genhd.c
+++ b/block/genhd.c
@@ -920,7 +920,6 @@ void del_gendisk(struct gendisk *disk)
 	bdev = bdget_disk(disk, 0);
 	if (bdev)
 		mutex_lock(&bdev->bd_mutex);
-	disk->flags &= ~GENHD_FL_UP;
 	/* invalidate stuff */
 	disk_part_iter_init(&piter, disk,
 			     DISK_PITER_INCL_EMPTY | DISK_PITER_REVERSE);
@@ -936,6 +935,7 @@ void del_gendisk(struct gendisk *disk)
 
 	invalidate_partition(disk, 0);
 	set_capacity(disk, 0);
+	disk->flags &= ~GENHD_FL_UP;
 	up_write(&disk->lookup_sem);
 
 	if (!(disk->flags & GENHD_FL_HIDDEN))
-- 
2.26.2

